### DISCUSSION TITLE
Can't signin to GitHub

### DISCUSSION THREAD

**Discussion by starmadegeek** [gitpod-io/openvscode-server#127] [Q&A] [answered]
Hi, I installed openvscode-server behind apache2 proxy on a subdomain and it works fine but I can't log in using GitHub into vscode. It shows error.
![image](https://user-images.githubusercontent.com/83012159/135580416-5e9cee16-3820-4490-95ad-bd3a2858ee72.png)
How to make this work? It's important for me as it syncs my extensions and settings.

**Comment by starmadegeek** [comment]
> Hi, I installed openvscode-server behind apache2 proxy on a subdomain and it works fine but I can't log in using GitHub into vscode. It shows error. ![image](https://user-images.githubusercontent.com/83012159/135580416-5e9cee16-3820-4490-95ad-bd3a2858ee72.png) How to make this work? It's important for me as it syncs my extensions and settings.

No one faced this problem? If its only me, then please help me out how to solve this or tell me what could be the problem.

**Reply by akosyakov** [reply]
Please follow this issue: https://github.com/gitpod-io/openvscode-server/issues/29

Microsoft allows access to these services only from MS products. We are working on the alternative right now by reusing Gitpod setting sync server.

**Comment by filiptronicek** [comment]
Thinking of ways around this, because this looks like a pretty important part of many users' dev flow, maybe we could implement https://github.com/VSCodium/vscodium/blob/master/patches/use-github-pat.patch for sign in via PAT? cc @akosyakov @jeanp413

**Reply by akosyakov** [reply]
I think we should raise an issue in upstream first. They used to allow to provide a token via a command. It makes sense to reintroduce to support standalone server. Alternatively ask why it does not work with forks? Maybe it is simply we forgot to configure something.

What we should no do is to change MS code lightly. We aim to easy catch up effort.

**Reply by filiptronicek** [reply]
It looks like it should be working now, I triggered an insiders release, which I'll test as soon as it completes. relevant commit: https://github.com/microsoft/vscode/commit/e8a26d3ebf5095d4324ad4bf794787540ed3b33e

**Comment by filiptronicek** [comment, marked as answer]
Since https://github.com/gitpod-io/openvscode-server/releases/tag/openvscode-server-insiders-v1.64.0 this issue is now resolved. You 

Fixing commit: https://github.com/microsoft/vscode/commit/f67a8b753f55e452f38b61497662839c4691fb12
Tracking issue: https://github.com/microsoft/vscode/issues/139253 and https://github.com/microsoft/vscode/issues/139566 (thanks @abhijeet-1110 for raising it!)

**Reply by filiptronicek** [reply]
The auth process is done via a device code: 
![image](https://user-images.githubusercontent.com/29888641/146973432-81b020f8-b0f6-4f67-a46b-960ef650beb9.png)
![image](https://user-images.githubusercontent.com/29888641/146973460-9c52424e-6b2d-4e7f-b6a5-c5c2313c53a5.png)

**Reply by abhijeet-1110** [reply]
@filiptronicek Thank you

**Reply by akosyakov** [reply]
good find, interesting GH api too, it was not there like a year ago

**Reply by bhavitsharma** [reply]
Not sure if this has regressed. I don't get the auth process via device code. I clicked on "Sign In" from GitHub extension. I am on 1.66.1.

**Reply by jeanp413** [reply]
@bhavitsharma please use 1.65.2 in the meantime, there's a bug in 1.66.x version that's affecting built-in extensions in this case github-auth extension that could be related

**Reply by bhavitsharma** [reply]
Thanks, that worked!

**Reply by YepizTechJonathan** [reply]
Hello, can someone please tell me what is the sign in process please and if it is still working on version 1.67?

**Reply by jeanp413** [reply]
Hi @YepizTech-Jonathan if you for example are using the Github Pull Request extension, it will ask you to login. It's working.
If you were using 1.66.x or 1.67.0 before please update to 1.67.1 and before running the server delete all extension from `~/.openvscode-server/extensions` folder
![image](https://user-images.githubusercontent.com/25115070/169055713-0058c68e-b9c9-40aa-9ffc-329c4317f8ef.png)

**Reply by YepizTechJonathan** [reply]
i don't get that sign in button on my instance
![chrome_6SEUweAyQe](https://user-images.githubusercontent.com/62561353/169256346-ebfb07c9-a319-4f4a-9ef4-1460b63fcc70.png)
![chrome_IX9dOEKFdn](https://user-images.githubusercontent.com/62561353/169256364-8051158e-552d-4137-9cb5-1033a1b2c10e.png)

**Reply by YepizTechJonathan** [reply]
i deleted the extensions folder and re installed just that one and nothing

**Reply by jeanp413** [reply]
@YepizTech-Jonathan As you see in the message you need to open a folder first that has git initialized

**Reply by YepizTechJonathan** [reply]
got it thank you
